### PR TITLE
Improve progress bar to sort by status and to have option to hide finished tasks

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
Helps de-clutter the progress bar and show the most important items first

Screenshot 2023-07-26 at 11 23 39 AM

Screenshot 2023-07-26 at 11 23 45 AM

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by chaowanggg** [comment]
Looks like the DCO is missing

File: dashboard/client/src/components/ProgressBar/ProgressBar.tsx [hunk 1]
```diff
@@ -169,31 +174,41 @@ export const ProgressBar = ({
...
+                    {label}: {value}
+                  </Typography>
+                  {hint && <HelpInfo className={classes.hint}>{hint}</HelpInfo>}
+                </div>
+              ))}
             </div>
-          ))}
-        </div>
+          )}
+          {controls && controls}
```
Comment (chaowanggg) [inline dashboard/client/src/components/ProgressBar/ProgressBar.tsx:208]: Nice!

**Review by chaowanggg — APPROVED** [review]
LGTM!

**Comment by scottsun94** [comment]
LGTM. The checkbox might be a bit hard to notice. However, it's probably fine for now. The sorting should be enough in most of the cases and we can adjust it later

**Pull request merged**